content/api/substitutions-reference.apib

```diff
@@ -712,47 +760,95 @@ Finally, as a more realistic example, `render_dynamic_content()` can also be use
 
-Example:
+```
+{
+  "id" : "ourfooter",
+  "content" : {
+    "html" : "<footer><p>Our standard html footer content</p></footer>",
+    "text" : "Our standard plain text footer content",
+    "amp_html" : "<footer><p>Our standard amp html footer content</p></footer>"
+  }
+}
+```
 
-```html
-{{ if not empty(shopping_cart) }}
-<table border = "1">
-<tr>
-<th>Name</th>
-<th>Price</th>
-</tr>
-{{ each shopping_cart }}
-<tr>
-<td>{{loop_var.item_name}}</td>
-<td>${{loop_var.item_price}}</td>
-</tr>
-{{ end }}
-</table>
-{{ else }}
-<b>Buy something!</b>
-{{ end }}
+The snippet can then be imported into either plain text, html, amp_html, or header email content
+using the `render_snippet` macro call. The macro call will automatically use the appropriate
+content.html, content.amp_html, or content.text snippet value based on the type of content that the snippet
+is being inserted into.
+For example, if a transmission was injected with content.html of the form:
+```
+<html>
+<p>Our body content</p>
+{{ render_snippet( "ourfooter" ) }}
+</html>
 ```
 
-**Braces Macros**
+The resulting rendered HTML email content would look like:
+```
+<html>
+<p>Our body content</p>
+<footer><p>Our standard html footer content</p></footer>
+</html>
+```
 
-The four macros for outputting braces are listed below followed by their output:
+The `render_snippet` macro takes a snippet ID as its only argument, which may come from `substitution_data`.
+The following example illustrates how one of two plain text snippets can be utilized, based on the recipient's `substitution_data`.
 
-* `opening_double_curly()` - {{
-* `closing_double_curly()` - }}
-* `opening_triple_curly()` - {{{
-* `closing_triple_curly()` - }}}
+Two snippets are created:
+```
+{
+  "id" : "banner_snippet_A",
+  "content" : {
+    "text" : "Banner A"
+  }
+}
+```
+
+```
+{
+  "id" : "banner_snippet_B",
+  "content" : {
+    "text" : "Banner B"
+  }
+}
+```
+
+transmission with content.text:
+```
+The following banner depends on the "banner_id" substitution_data value
+{{ render_snippet( banner_id ) }}
+```
 
+Where Recipient 1 has substitution_data:
+```
+{
+  "banner_id" : "banner_snippet_A"
+}
+```
+
+And Recipient 2 has substitution_data:
+```
+{
+  "banner_id" : "banner_snippet_B"
+}
+```
 
+Other notes on snippet usage:
+* Snippets themselves may contain substitution syntax. Though some restrictions apply: snippets cannot reference other snippets (`render_snippet`), nor can they utilize `render_dynamic_content`.
+* Snippets may contain links (which will be click tracked if `click_tracking` is enabled).
+* If a `render_snippet` call references a snippet which does not exist, the transmission will be discarded with a generation failure event.
+* Each of the HTML, AMPHTML, and plain text email parts may utilize `render_snippet` at most 30 times. If this limit is exceeded, the transmission will be discarded with a generation failure event.
```

Comment:
Used at most 30x *per* transmission call, right (not 30x period)?

Reply:
30 `render_snippet()` calls per part. So text, html, and amp_html can each have 30 calls in them.